### PR TITLE
create battle by react

### PR DESCRIPTION
**Pull request by DeborahAlmeida**
## Objective and Motivation
- Create Battle by React Component
- Using Formik Library to render the form

### CODE REVIEW

File: frontend/js/pages/BattleCreate.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  useEffect(() => {
+    if (!user) {
+      props.getCurrentUser();
+    }
+  }, []);
+
+  if (user) {
+    return (
+      <div className="battle_container_detail">
+        <Formik
```
Comment (rsarai) [inline frontend/js/pages/BattleCreate.js:31]: let's add some validations of this form? You can use https://github.com/jquense/yup or use the default validations provided by formik

File: frontend/js/utils/api.js [hunk 2]
```diff
@@ -31,4 +31,56 @@ const getBattles = async () => {
...
+    for (const element of cookies) {
+      const cookie = element.trim();
+      if (cookie.slice(0, Math.max(0, name.length + 1)) === `${name}=`) {
+        cookieValue = decodeURIComponent(cookie.slice(Math.max(0, name.length + 1)));
+        break;
+      }
+    }
+  }
+  return cookieValue;
+};
```
Comment (rsarai) [inline frontend/js/utils/api.js:84]: Hmmm, how did you come to this?

**Review by rsarai — CHANGES_REQUESTED** [review]

File: frontend/js/utils/api.js (hunk 2 above)
Comment (DeborahAlmeida) [reply frontend/js/utils/api.js:84]: As we talked last daily meeting, I use this for getting token

File: frontend/js/pages/BattleCreate.js (hunk 1 above)
Comment (DeborahAlmeida) [reply frontend/js/pages/BattleCreate.js:31]: great!

**Review by rsarai — APPROVED** [review]

**Pull request merged**